### PR TITLE
Add immutable docs

### PR DESCRIPTION
**Pull request by jdlehman**
- Add documentation on examples of using Immutable.js with alt
- Create folder structure to add more docs on using other immutable libs

### CODE REVIEW

**Comment by jdlehman** [comment]
Would like a quick read over to make sure these make sense.

File: docs/immutableExamples/immutablejs.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
+# Using Immutable Data Structures
```
Comment (goatslacker) [inline docs/immutableExamples/immutablejs.md]: can we change the directory name to just `immutable`?

**Comment by goatslacker** [comment]
We should add a note here on serializing/deserializing immutable data.

**Comment by goatslacker** [comment]
Otherwise this is :+1:

File: docs/immutable/immutablejs.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+// MyStore.js
+import {Map} from 'immutable';
+
+class MyStore {
+  constructor() {
+
+    this.on('serialize', () => {
+      return {
+        data: {
+          prop1: this.data.get('prop1'),
```
Comment (goatslacker) [inline docs/immutable/immutablejs.md]: can't you use fromJS/toJS here or no?

Comment (jdlehman) [inline docs/immutable/immutablejs.md]: Yep, that is actually much cleaner. I will update that. Thanks :)

**Pull request merged**